functions/Get-DbaDiagnosticInfo.ps1

```diff
@@ -0,0 +1,129 @@
+Function Get-DbaDiagnosticInfo
+{
+<#
+.SYNOPSIS
+Collects information about environment where DBAtools is being executed from.
+
+.DESCRIPTION
+Collects information about environment where DBAtools is being executed from. The information will be helpful for troubleshooting issues reported by users.
+    
+
+.PARAMETER Detailed
+Capture detailed information.
+
+.NOTES 
+Original Author: Chrissy LeMaire
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Get-DbaDiagnosticInfo
+
+.EXAMPLE
+Get-DbaDiagnosticInfo
+
+This will collect basic information about the computer where DBAtools is being executed from. It will then show the information in the conslole and at the same time COPY it. After running the cmdlet just paste it where ever you are reporting the issue: GitHub issues, email, or SQLcommunity slack channel etc.
+
+.EXAMPLE   
+Get-DbatoolsEnvironmentInfo -Detailed
+
+This will show MORE DETAILED information about the computer where DBAtools is being executed from.
+    
+#>
+    Param (
+        #[string]$ComputerName = "$ENV:COMPUTERNAME",
+        [switch]$Detailed,
+        [switch]$Clip
+    )
+    
+    try
+    {
+        
+        # #Write-Verbose "Collecting information about the computer where DBAtools is excuted from"
+
+        Write-Verbose "Getting local enivornment information"
+        $localinfo = @{ } | Select-Object OSVersion, OsArchitecture,PowerShellVersion, PowerShellArchitecture, DbaToolsVersion, ModuleBase, CLR, SMO, DomainUser, RunAsAdmin, isPowerShellISE
+        
+        $localinfo.OSversion = [environment]::OSVersion.Version.ToString() 
+        $OsVersion = (Get-CimInstance CIM_OperatingSystem).Caption.ToString()
```

Comment:
I think it would be best to test if remoting works first. It's a useful addition to this function and WinRM is a prerequisite for Get-Ciminstance, even on localhost.